src/NHibernate.Test/Extralazy/ExtraLazyFixture.cs

```diff
@@ -23,12 +26,2206 @@ protected override string CacheConcurrencyStrategy
+				}
+
+				Assert.That(gavin.Companies.Count, Is.EqualTo(10));
+				Assert.That(Sfi.Statistics.FlushCount, Is.EqualTo(0));
+				Assert.That(Sfi.Statistics.PrepareStatementCount, Is.EqualTo(0));
+				Assert.That(NHibernateUtil.IsInitialized(gavin.Companies), Is.False);
+
+				// Test inserting companies at the end
+				Sfi.Statistics.Clear();
+				for (var i = 10; i < 15; i++)
+				{
+					var item = new Company($"c{i}", i, gavin);
+					addedItems.Add(item);
+					gavin.Companies.Insert(i, item);
+				}
+
+				Assert.That(gavin.Companies.Count, Is.EqualTo(15));
+				Assert.That(Sfi.Statistics.FlushCount, Is.EqualTo(0));
+				Assert.That(Sfi.Statistics.PrepareStatementCount, Is.EqualTo(0));
+				Assert.That(NHibernateUtil.IsInitialized(gavin.Companies), Is.False);
+
+				// Try insert invalid indexes
+				Assert.Throws<ArgumentOutOfRangeException>(() => gavin.Companies.RemoveAt(-1));
+				Assert.Throws<ArgumentOutOfRangeException>(() => gavin.Companies.RemoveAt(20));
+
+				// Check existance of added companies
+				Sfi.Statistics.Clear();
+				foreach (var item in addedItems.Skip(5))
+				{
+					Assert.That(gavin.Companies.Contains(item), Is.True);
+				}
+
+				Assert.That(Sfi.Statistics.FlushCount, Is.EqualTo(0));
+				Assert.That(Sfi.Statistics.PrepareStatementCount, Is.EqualTo(0));
+				Assert.That(NHibernateUtil.IsInitialized(gavin.Companies), Is.False);
+
+				// Check existance of not loaded companies
+				Assert.That(gavin.Companies.Contains(addedItems[0]), Is.True);
+				Assert.That(gavin.Companies.Contains(addedItems[1]), Is.True);
+
+				Assert.That(Sfi.Statistics.FlushCount, Is.EqualTo(0));
+				Assert.That(Sfi.Statistics.PrepareStatementCount, Is.EqualTo(2));
+				Assert.That(NHibernateUtil.IsInitialized(gavin.Companies), Is.False);
+
+				// Check existance of not existing companies
+				Sfi.Statistics.Clear();
+				Assert.That(gavin.Companies.Contains(new Company("test1", 15, gavin)), Is.False);
+				Assert.That(gavin.Companies.Contains(new Company("test2", 16, gavin)), Is.False);
+
+				Assert.That(Sfi.Statistics.FlushCount, Is.EqualTo(0));
+				Assert.That(Sfi.Statistics.PrepareStatementCount, Is.EqualTo(2));
+				Assert.That(NHibernateUtil.IsInitialized(gavin.Companies), Is.False);
+
+				if (initialize)
+				{
+					using (var e = gavin.Companies.GetEnumerator())
+					{
+						e.MoveNext();
+					}
+
+					Assert.That(NHibernateUtil.IsInitialized(gavin.Companies), Is.True);
+					Assert.That(gavin.Companies.Count, Is.EqualTo(15));
+				}
+
+				t.Commit();
+			}
+
+			using (var s = OpenSession())
+			using (var t = s.BeginTransaction())
+			{
+				gavin = s.Get<User>("gavin");
+				Assert.That(gavin.Companies.Count, Is.EqualTo(15));
+				Assert.That(NHibernateUtil.IsInitialized(gavin.Companies), Is.False);
+
+				t.Commit();
+			}
+		}
+
+		[TestCase(false, false)]
+		[TestCase(false, true)]
+		[TestCase(true, false)]
+		[TestCase(true, true)]
+		public void ListInsertDuplicated(bool initialize, bool flush)
```

Comment:
I would also set this as explicitly run only.

Reply:
Marked as `Explicit`.